modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/IgniteH2Indexing.java

```diff
@@ -2194,6 +2194,68 @@ && matches(QueryUtils.sysSchemaName(), schemaNamePtrn)) {
         markIndexRebuild(cctx.name(), true);
     }
 
+    /** {@inheritDoc} */
+    @Override public @Nullable IgniteInternalFuture<?> rebuildInMemoryIndexes(GridCacheContext cctx) {
+        assert cctx != null;
+
+        if (ctx.clientNode())
+            return null;
+
+        IgnitePageStoreManager pageStore = cctx.shared().pageStore();
+
+        if (pageStore == null)
```

Comment:
This is interesting, why do you have to check page store existence and not use it later? Please add a comment in the code.

Reply:
Added comment in code